magento/magento2
# MSI-1542: Provide MSI support for Shipment Web API endpoint

### Description
The problem is, there is no data migration from request extension attributes to shipment extension attributes. I added service for migration request extension attributes data to shipping extension attributes data 

### Fixed Issues (if relevant)
1. [magento-engcom/msi-1542](https://github.com/magento-engcom/msi/issues/1542): Provide MSI support for Shipment Web API endpoint

### Manual testing scenarios
1. Go to Magento/Catalog/etc/extension_attributes.xml and add extension attribute like this 
```
    <extension_attributes for="Magento\Sales\Api\Data\ShipmentInterface">
        <attribute code="test_attribute_name" type="string" />
    </extension_attributes>

    <extension_attributes for="Magento\Sales\Api\Data\ShipmentCreationArgumentsInterface">
        <attribute code="test_attribute_name" type="string" />
    </extension_attributes>
```
2. Flush cache bin/magento cache:flush
3. Run command bin/magento setup:di:compile
4. Create order via storefront
5. Add breakpoint to Magento/Sales/Model/Order/ShipmentDocumentFactory.php:91 like this http://prntscr.com/kfcpqg
6. Run POST request /V1/order/:orderId/ship
_example: http://example.com/rest/V1/order/1/ship_
with body 
`{
  "arguments": {
    "extension_attributes": {
      "test_attribute_name": "test_attribute"
    }
  }
}`
7. Check that the shipping has extension attribute http://prntscr.com/kfcxim


### Contribution checklist
